R-package/R/create-release-files.R

```diff
@@ -0,0 +1,411 @@
+    errors <- FALSE
+    for (row_index in seq_len(nrow(files_sheet_formatted))) {
+      working_row <- files_sheet_formatted[row_index,]
+      
+      # File extraction info
+      fileID <- working_row[[files$file_id$name]]
+      file_name <- working_row[[files$file_name$name]]
+      file_type <- working_row[[files$file_type$name]]
+      partID <- working_row[[files$part_ID$name]]
+      add_headers <- working_row[[files$add_headers$name]]
+      destination <- working_row[[files$destinations$name]]
+      osf_location <- working_row[[files$osf_locations$name]]
+      github_location <- working_row[[files$github_location$name]]
+      
+      validated_file <- FALSE
+      
+      # Validate destination
+      if (!(destination %in% files$destinations$categories)) {
+        logger::log_warn(paste0("File ID: ",
+                                fileID,
+                                " has an invalid destination set"))
+        errors <- TRUE
+      }
+      
+      set_info <- sets_sheet[sets_sheet$setID == partID, ]
+      # Determine if the partID supplied is set or part.
+      if (file_type == files$file_type$categories$excel) {
+        # If part exists in sets its a set therefore all parts belonging to the set are used as partID for sheet creation.
+        if (nrow(set_info) >= 1) {
+          set_name <- partID
+          set_parts <- set_info[["partID"]]
+          partID <- set_parts
+          # Set names for elements to allow removal of invalid parts
+          names(set_parts) <- set_parts
+          for (single_part in partID) {
+            if (single_part %in% parts_sheet[["partID"]]) {
+              # Check that a sheet with this part exists
+              if (single_part %in% names(dictionary)) {
+                next()
+              } else{
+                logger::log_warn(
+                  paste0(
+                    single_part,
+                    " does not have a matching sheet but is part of ",
+                    set_name,
+                    " set, which was selected to be exported."
+                  )
+                )
+                errors <- TRUE
+                # Remove missing part
+                set_parts <-
+                  set_parts[names(set_parts) != single_part]
+              }
+              
+            } else{
+              logger::log_warn(paste0(single_part,
+                                      " is missing from the parts sheet but is present in the ",
+                                      set_name,
+                                      " set."))
+              errors <- TRUE
+              # Remove missing part
+              set_parts <-
+                set_parts[names(set_parts) != single_part]
+            }
+          }
+          partID <- unname(set_parts)
+          # Check if any valid parts remain
+          if (length(partID) >= 1) {
+            validated_file <- TRUE
+          }
+        } else{
+          if (partID %in% parts_sheet[["partID"]]) {
```

Comment:
This line and the next should be put into a function which checks if a partID is a valid part from the parts sheet. It also being used when the fileType is a CSV.